File: Crio/Recursion/5_NQueens/NQueens.java
import java.util.*;

public class NQueens {
    //CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    String answer;
    void place(int index , int[] column , int n) {
       if(index == n) {
            for(int i = 0 ; i < n ; i++) {
                for(int j = 0 ; j < n ; j++) {
                    if(column[j] == i) {
                        answer += '1';
                    } else {
                        answer += '0';
                    }
                }
                answer += '\n';
            }
            answer += '\n';
            return;
        }
        for(int i = 0 ; i < n ; i++) {
            boolean canPlace = true;
            for(int j = 0 ; j < index ; j++) {
                if(column[j] == i) {
                    canPlace = false;
                    break;
                }
                if(Math.abs(column[j] - i) == Math.abs(index-j)) {
                    canPlace = false;
                    break;
                }
            }
            if(canPlace) {
                column[index] = i;
                place(index+1, column , n);
            }
        }
    }
    //CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    String nQueens(int n) {
        //CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        answer = new String();
        int column[] = new int[n];
        place(0 , column , n);
        boolean isAns = false;
        for(int i = 0 ; i < answer.length() ; i++) {
            if(answer.charAt(i) == '1') {
                isAns = true;
            }
        }
        if(isAns == false) {
            return "No Solution Exists";
        }
        return answer;
        //CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n =  scanner.nextInt();
        scanner.close();
        NQueens result = new NQueens();
        String board = result.nQueens(n);
        System.out.println(board);
    }
}

File: Crio/Recursion/5_NQueens/NQueens.cpp
#include<bits/stdc++.h>
using namespace std;
//CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
int board[10][10];
string answer;
void concatBoard(int n)
{
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			string c=to_string(board[i][j]);
			answer+=c;
		}
        answer.append("\n");
	}
	answer.append("\n");
}
bool check(int row, int col,int n)
{
	int i, j;

	for (i=0;i< col;i++)
		if (board[row][i])
			return false;

	for (i=row,j=col;i>=0&&j>=0;i--,j--)
		if (board[i][j])
			return false;

	for (i=row,j=col;j>=0&&i<n;i++,j--)
		if (board[i][j])
			return false;

	return true;
}

bool solve(int col,int n)
{
	if (col == n)
	{
		concatBoard(n);
		return true;
	}
	bool res = false;
	for (int i = 0; i <n; i++)
	{
		if ( check(i,col,n) )
		{
			board[i][col] = 1;
			res = solve(col + 1,n) || res;
			board[i][col] = 0;
		}
	}
	return res;
}
//CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
string nQueens(int n)
{
	//CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	bool t=solve(0,n);
	if(t==true)
		return answer;
	else
		return "No Solution Exists\n";
	//CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main()
{
    int n;
    cin>>n;
	cout<<nQueens(n);
}
